File: java/src/test/java/com/bc/libwally/util/TestUtils.java
package com.bc.libwally.util;

public class TestUtils {

    @SuppressWarnings("unchecked")
    public static <T extends Throwable> T assertThrows(String msg,
                                                       Class<T> expectedError,
                                                       Callable callable) {
        try {
            callable.call();
            throw new AssertionError(msg);
        } catch (Throwable e) {
            if (expectedError.isAssignableFrom(e.getClass()))
                return (T) e;
            throw e;
        }
    }
}

File: android/app/src/androidTest/java/com/bc/libwally/util/TestUtils.kt
package com.bc.libwally.util

inline fun <reified T : Throwable> assertThrows(
    msg: String?,
    callable: () -> Unit
): T {
    try {
        callable()
        throw AssertionError(msg)
    } catch (e: Throwable) {
        if (e is T) return e
        throw e
    }
}